src/services/api/contract_class.rs

```diff
@@ -93,3 +112,100 @@ impl ContractClass {
         Ok(())
     }
 }
+
+// -------------------
+//  Helper Functions
+// -------------------
+
+impl TryFrom<PathBuf> for ContractClass {
+    type Error = io::Error;
+
+    fn try_from(path: PathBuf) -> io::Result<Self> {
+        let file = File::open(path)?;
+        let reader = BufReader::new(file);
+        let raw_contract_class: starknet_api::state::ContractClass =
+            serde_json::from_reader(reader)?;
+
+        let contract_class = ContractClass::from(raw_contract_class);
+        Ok(contract_class)
+    }
+}
+
+fn convert_entry_points(
+    entry_points: HashMap<starknet_api::state::EntryPointType, Vec<EntryPoint>>,
+) -> HashMap<EntryPointType, Vec<ContractEntryPoint>> {
+    let mut converted_entries: HashMap<EntryPointType, Vec<ContractEntryPoint>> = HashMap::new();
+    for (entry_type, vec) in entry_points {
+        let en_type = match entry_type {
+            starknet_api::state::EntryPointType::Constructor => {
+                crate::services::api::contract_class::EntryPointType::Constructor
```

Comment:
Do you need to use the absolute path? I'm not sure if there's a conflict, but if there is maybe you could use a type alias
```rs
type TheirEPT = starknet_api::state::EntryPointType;
type OurEPT = crate::services::api::contract_class::EntryPointType;

match entry_type {
    TheirEPT::Constructor => OurEPT::Constructor,
    // ...
}
```
or even better:
```rs
// Where our EntryPointType is defined:
impl From<starknet_api::state::EntryPointType> for EntryPointType {
    // convert with typealiases as before
}

// Then in the pinned line, you just need to do this:
entry_type.into()
```

Reply:
there was a conflict because both enums  were called equally, so i had to use the paths to differentiate them. I used both of your suggestions, thanks for pointing them out.